lib/cfg.c

```diff
@@ -504,6 +504,11 @@ cfg_new(gint version)
 
   self->use_uniqid = FALSE;
 
+  /* The rcptid part of use_uniqid() is used in trace messages, enable it
+   * explicitly.  We don't care about performance if trace is enabled.  */
+  if (trace_flag)
+    self->use_uniqid = TRUE;
```

Comment:
The trace flag can be enabled at runtime using syslog-ng-ctl as well. In that specific case, rcptid will not work.

Reply:
I know about that, but I felt that the more common use-case is to start syslog-ng interactively in foreground mode. And rcptid can only initialized at startup at the moment. Not saying that it couldn't be done, but I was fine with this restriction.